aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -44,11 +44,25 @@ static inline int64_t matrixStride(const Tensor& batched_matrices) {
   return batched_matrices.size(-1) * batched_matrices.size(-2);
 }
 
+/*
+ * Given batches of matrices with arbitrary batch dim,
+ * computes the number of batches for Triu and Tril. This ignores stride 0 dimension
+ */
+static inline int64_t batchCountTrilTriu(const Tensor& batched_matrices) {
+  int64_t result = 1;
+  for (int64_t i = 0; i < batched_matrices.ndimension() - 2; i++) {
+    if (batched_matrices.stride(i) != 0) {
+      result *= batched_matrices.size(i);
+    }
+  }
+  return result;
+}
+
 /* Checks a necessary property for the triu and tril implementations, hence the name.
  * Here batch contiguity is checked for tensors with greater than 4 dimensions.
  * Contiguous tensors and tensors with less than 3 dimensions pass this check
  */ 
-static inline std::tuple<bool, Tensor> checkTrilTriuBatchContiguous(const Tensor& tensor) {
+static inline std::tuple<bool, Tensor> checkTrilTriuBatchContiguous(const Tensor& tensor, bool allow_zero_stride) {
   // Complete contiguity is the most desired property, which is why
   // we return true if the tensor is contiguous
   if (tensor.is_contiguous()) {
```

Comment:
Do we ever have situation when is_contiguous tensor doesn't satisfy `tensor.strides() == default_strides_for_size`. As I can see `defaultStrides` and `conmute_contiguous` using exactly same strides formula.

Reply:
Here we go. :)
```
In [7]: x = torch.randn(2, 10, 2)                                                                                                    

In [8]: y = x[::2, :, :]                                                                                                             

In [9]: y.is_contiguous()                                                                                                            
Out[9]: True

In [10]: y.size()                                                                                                                    
Out[10]: torch.Size([1, 10, 2])

In [11]: y.stride()                                                                                                                  
Out[11]: (40, 2, 1)```

Reply:
dimension with size 1 always got me as well